roles/docker/tasks/main.yml

```diff
@@ -47,12 +47,14 @@
   file:
     path: "{{ docker_systemd_dir }}"
     state: directory
+  when: not os_firewall_use_firewalld | default(True) | bool
 
 # Extend the default Docker service unit file
 - name: Configure Docker service unit file
   template:
     dest: "{{ docker_systemd_dir }}/custom.conf"
     src: custom.conf.j2
+  when: not os_firewall_use_firewalld | default(True) | bool
```

Comment:
Lets get these two tasks moved into a block and apply the conditional to the block instead of duplicating the when clause.